File: 03- Sliding Window/C#/1- Maximum Average Subarray I.cs
public class Solution {
    public double FindMaxAverage(int[] nums, int k) {
        double sum = 0 , average = 0; 
        for (int i = 0 ; i < k ; i++)
            sum += nums[i] ; 
            
        average = Math.Round(sum / k , 5) ;
        for (int i = k ; i < nums.Length ; i++){
            sum -= nums[i-k];
            sum += nums[i] ; 
            average = Math.Max(average , Math.Round(sum / k , 5));
        }
        return average ; 
    }
}

File: 03- Sliding Window/1- Maximum Average Subarray I.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        double sum = 0, average = 0;
        for (int i = 0; i < k; i++)
            sum += nums[i];
        average = round((sum / k) * 1e5) / 1e5;
        for (int i = k; i < nums.size(); i++) {
            sum -= nums[i - k];
            sum += nums[i];
            average = max(average, round((sum / k) * 1e5) / 1e5);
        }
        return average;
    }
};
